lib/bamboo/api_error.ex

```diff
@@ -39,8 +39,10 @@ defmodule Bamboo.ApiError do
 
       <extra_message>
   """
+  @spec raise_api_error(any()) :: no_return()
   def raise_api_error(message), do: raise(__MODULE__, message: message)
 
+  @spec raise_api_error(String.t(), any(), Keyword.t()) :: no_return()
```

Comment:
I'm not 100% sure that the third argument will always be a `Keyword.t()`. I imagine it can also be a map. Could we relax that to an `Enum.t()`, since that's a more relaxed version that includes both keyword lists and maps? 

```suggestion
  @spec raise_api_error(String.t(), any(), Enum.t()) :: no_return()
```